src/react/components/StyleEditor/Preview.jsx

```diff
@@ -24,109 +25,37 @@ const areaPreviewSVG = '<svg viewBox="0 0 80 80" width="80" height="80" xmlns="h
-            strokeLineCap: defaults.stroke.lineCap,
-            fill: defaults.defaultFill, // tyhjä stringi tähän?
-            fillColor: defaults.fill.color,
-            pattern: defaults.fill.area.pattern
-        };
-    }
-    
-    _fillSvgWithStyle () {
-        const format = this.props.format;
-        const path = this._parsePath(format);        
-
-        this.previewAttributes.strokeColor = format === 'area' ? this.props.oskariStyle.stroke.area.color : this.props.oskariStyle.stroke.color;
-        this.previewAttributes.fillColor = this.props.oskariStyle.fill.color;
-
-        this.previewAttributes.fill = format === 'area'
-            ? ('url(#' + defaultPatternId + ')') : format !== 'line' 
-            ? this.props.oskariStyle.image.fill.color : defaults.fill.color;
-
-        if (format === 'area' && ~this.props.oskariStyle.fill.area.pattern) {
-            const patternPath = this._parsePattern(this.props.areaFills.find(pattern => pattern.name === this.props.oskariStyle.fill.area.pattern));
-            this.previewAttributes.pattern = this._composeSvgPattern(patternPath);
-        }
-
-        this.previewAttributes.strokeWidth = format === 'point'
-            ? defaults.defaultStrokeWidth : format === 'area'
-            ? this.props.oskariStyle.stroke.area.width : this.props.oskariStyle.stroke.width;
-        
-        this.previewAttributes.strokeLineCap = format === 'line' ? this.props.oskariStyle.stroke.lineCap : defaults.stroke.lineCap;
-
-        this.previewAttributes.strokeDashArray = this._getPreviewLineDash(format, this.props.oskariStyle);
-
-        this.previewAttributes.strokeLineJoin = format === 'line' ? this.props.oskariStyle.stroke.area.lineJoin : defaults.stroke.area.lineJoin;
-
-        path.setAttribute('stroke', this.previewAttributes.strokeColor);
-        path.setAttribute('stroke-width', this.previewAttributes.strokeWidth);
-        path.setAttribute('stroke-linecap', this.previewAttributes.strokeLineCap);
-        path.setAttribute('stroke-dasharray', this.previewAttributes.strokeDashArray);
-        path.setAttribute('stroke-linejoin', this.previewAttributes.strokeLineJoin);
-        if (format !== 'line') {
-            path.setAttribute('fill', this.previewAttributes.fill);
-        }
-        
-        this.size = format === 'point' ? this.props.oskariStyle.image.size : defaults.image.size;
-
-        return path.outerHTML;
-    }
+export const Preview = (props) => {
+    const {
+        markers,
+        areaFills,
+        format,
+        oskariStyle
+    } = props;
+
+    let size = 3;
+
+    let previewAttributes = {
+        strokeColor: PREVIEW_DEFAULTS.stroke.color,
+        strokeWidth: PREVIEW_DEFAULTS.stroke.width,
+        strokeLineCap: PREVIEW_DEFAULTS.stroke.lineCap,
+        fill: PREVIEW_DEFAULTS.defaultFill, // empty string here?
+        fillColor: PREVIEW_DEFAULTS.fill.color,
+        pattern: PREVIEW_DEFAULTS.fill.area.pattern
+    };
```

Comment:
Most of these helper functions and variables don't need to be inside the render function. Move them out.

Reply:
Moved.